### DISCUSSION TITLE
OLED Display

### DISCUSSION THREAD

**Discussion by michaelhsapp** [micbay/RaceTimerAndController#21] [Ideas]
The combination of LED & LCD displays work well and conserve the I/O pin utilization of the Arduino effectively. Maybe consider incorporating an OLED display into a future build. OLED's are relatively inexpensive, are I/O pin utilization friendly and very versatile at displaying text and other characters. The downside is the displays are kind of small, at least the ones I'm familiar with.

**Comment by micbay** [comment]
OLEDs look great, for a version where cost is not a factor, one could certainly be used.

However, as you note, the inexpensive hobby OLEDs are very, small, and once you get to a physical size equal to, or bigger than a character LCD it gets to be at least 2-3X the LCD, and then exponentially more expensive as size increases from there.
Beyond just being a user interface, the main display is intended to be a spectator viewable leader board, so after the minimum necessary UI information, viewability from a distance was the most important design factor.

Another factor to consider, other than cost/size, is the amount of memory required to store and draw the screens.
I haven't made a direct comparison, but since OLED has more graphical potential, and many more pixels, I suspect it would also require more code to interface with, and thus more program memory, as well as more data storage to hold graphical design elements that can't be programmatically drawn.

Since adding drag race functionality we've pretty much reached the memory limit of the Arduino Nano, so a board with more memory, or some feature trade-offs may be necessary for any changes that require more bytes of code. That said, some memory can be immediately recovered if not using the Adafruit start light (which most people aren't), by removing the adafruit libraries and adafruit start light code.
